templates/identity/email-password/register.html.ejs

```diff
@@ -12,58 +12,184 @@
     <h1>Community Solid Server</h1>
   </header>
   <main>
-    <h1>Sign up for an account</h1>
-    <form action="/idp/register" method="post">
+    <h1>Sign up</h1>
+    <form action="/idp/register" method="post" id="mainForm">
+      <% const isBlankForm = !('email' in prefilled); %>
 
-      <%if (errorMessage) { %>
+      <% if (errorMessage) { %>
         <p class="error">Error: <%= errorMessage %></p>
       <% } %>
 
       <fieldset>
+        <legend>Your WebID</legend>
+        <p>
+          A <em>WebID</em> is a unique identifier for you
+          in the form of a URL.
+          <br>
+          You WebID lets you log in to Solid apps
+          and access non-public data in Pods.
+        </p>
         <ol>
-          <li>
-            <label for="email">Email</label>
-            <input id="email" type="text" name="email" autofocus <% if (prefilled.email) { %> value="<%= prefilled.email %>" <% } %> />
+          <li class="radio">
+            <label>
+              <input type="radio" id="createWebIdOn" name="createWebId" value="on"<% if (isBlankForm || prefilled.createWebId) { %> checked<% } %>>
+              Create a new WebID for my Pod.
+            </label>
+            <p id="createWebIdForm">
+              Please also create a Pod below, since your WebID will be stored there.
+            </p>
           </li>
-          <li>
-            <label for="password">Password</label>
-            <input id="password" type="password" name="password" />
-          </li>
-          <li>
-            <label for="confirmPassword">Confirm password</label>
-            <input id="confirmPassword" type="password" name="confirmPassword" />
+          <li class="radio">
+            <label>
+              <input type="radio" id="createWebIdOff" name="createWebId" value=""<% if (!isBlankForm && !prefilled.createWebId) { %> checked<% } %>>
+              Use my existing WebID to access my Pod.
+            </label>
+            <ol id="existingWebIdForm">
+              <li>
+                <label for="webId">Existing WebID:</label>
+                <input id="webId" type="text" name="webId" value="<%= prefilled.webId ?? '' %>" />
+              </li>
+              <li class="checkbox">
+                <label><input type="checkbox" id="register" name="register"<% if (isBlankForm || prefilled.register) { %> checked<% } %>>Use my new account to log in with this WebID.</label>
```

Comment:
A few lines in this document are longer than the git diff box.

Reply:
I didn't always find good split points, will go through it again.